base/Dockerfile.ubi8

```diff
@@ -45,11 +45,13 @@ ENV CUB_CLASSPATH='"/usr/share/java/cp-base-new/*"'
 # Zulu openJDK
 ENV ZULU_OPENJDK="zulu-8-8.40.0.25"
 
+ENV PYTHON_VERSION="36-3.6.8"
```

Comment:
Any reason why we're pining the `3.6.8` portion here? Makes us future proof for when Redhat updates this package to a newer version.

Reply:
@andrewegel Do you mean why we are pinning to this specific version of 3.6, or pinning to a version at all? I guess the answer is the same either way, reproducibility. If we need to recreate the 5.4.0 images for some reason we need them to be the exact same thing we had already released. If we release 5.4.5 we don't want the version of python to change because it could cause some issue and would require more testing every time we do a patch release. I believe for patch releases we only to change what we explicitly change and not just pick up any other updates, even if it is just patch update to a library or resource we depend on.

Reply:
I'm wondering the reason why would want to keep it at `3.6.8`, even after redhat releases `python36-3.6.10` (at some point they will) - Seems unnecessary to me to pin it to `3.6.8` specifically, as you're now missing security updates from Redhat in the container, possibly having our customers get bugged by their security/compliance officers telling them "You're running a container with an un-fixed CVE that was fixed in python3>=3.6.8"

I don't even see a `python36` package at the Redhat8 package manifest: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/package_manifest/baseos-repository

Just python3, so I assume that changing `python3` => `python36` you're following the "provides" way of doing things. To whit, you should instead use the "provides" way of installing a package that "provides" a python36 interpreter thats compatible with the 3.6 ABI in the form of: 

`'python(abi) = 3.6'`

See in dnf/yum on CentOS-7:

```
[centos@ip-172-31-13-239 ~]$ sudo yum install 'python(abi) = 3.6'
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: d36uatko69830t.cloudfront.net
 * extras: d36uatko69830t.cloudfront.net
 * updates: d36uatko69830t.cloudfront.net
Resolving Dependencies
--> Running transaction check
---> Package python3.x86_64 0:3.6.8-10.el7 will be installed
```

This way you still ensure that python36, and only 36 is installed, but it'll take a newer version greater than `3.6.8` when it gets released.

My suggestion would be to drop the `-3.6.8` on the end so future builds have the option to take a newer version.

If you're worried about compatibility changing between `3.6.8` and `3.6.9`, then thats valid if you have experiences in that area, but from my experience, Redhat & Python.org do a good job of remaining compatible between patch releases.

This isn't a hard blocker from me, just trying to make sure we follow best practices in terms of getting security updates.

Reply:
>  A lot of customers have requirements that nothing else can change except that small piece of code otherwise they can not deploy it. Pretty much one of the major reasons we even do hot fixes for people is so the version of our code and what we deliver doesn't change, other wise we would just say upgrade to the latest patch version of the release they are using which has the fix they need and others as well.

Thats a fair requirement - I went through this getting Amazon Linux's NSS package FIPS certified - Once certified, only CVE patches would be allowed in order to maintain FIPS certification. If we bumped the version, we'd have to go through the entire certification process again (time + $$). 

I've given this change my ship-it, and I'm not concerned.